src/Main.hs

```diff
@@ -180,3 +274,17 @@ main = do
             else raiseStatus status401 "Invalid Credentials"
     get "/" $ do
       json Color {peter = "peter", jonas = 3}
+    get "/recipes" $ json =<< liftIO (getAllRecipes db)
+    post "/recipes" $ do
+      recipe <- jsonData :: ActionM Recipe
+      success <- liftIO $ addRecipe db recipe
+      if success
+        then html "Recipe inserted into database!"
+        else raiseStatus status401 "Error occured on recipe insert!"
```

Comment:
Dieser fall tritt nie ein. Falls doch, sollte der server crashen.

Reply:
9d8ac164eaaf6e8355e1254ae071ac5eb29b6aef doesn't handle this exception anymore. Changed addRecipe's return type to () as it should never fail.